src/libexpr/nixexpr.hh

```diff
@@ -226,8 +227,9 @@ struct ExprVar : Expr
     Symbol name;
 
     /* Whether the variable comes from an environment (e.g. a rec, let
-       or function argument) or from a "with". */
-    bool fromWith;
+       or function argument) or from a "with", and if from a with its
+       source. */
```

Comment:
We don't really know statically which `with` it comes from, when `with`s are nested. IIUC the idea here is to link to the innermost, nearest `with`?

```suggestion
       or function argument) or from a "with".
       
       `nullptr`: Not from a `with`.
       Valid pointer: the nearest, innermost `with` expression to query first. */
```